### PR TITLE
BUILD: Update numpy requirement to >=1.20.0,<2.2

### PR DESCRIPTION
**Pull request by SMoraisAnsys**
As title says. Changes include:
- use `cdouble` instead of the long deprecated `complex_` when defining `dtype` with `numpy` functions;
- switch from `pytomlpp` to `toml`;
- update ci workflow to not use caching.

The reason for the switch in `toml` dependency is related to two facts:
1. There is a problem when combining Linux + `numpy>=2.0`+ `pytomlpp`. Indeed, in Linux, we end up loading the `libPyDesktopPlugin.so` file (see https://github.com/ansys/pyaedt/blob/c098962e49448358744a88dd01f3dc80be56918f/src/ansys/aedt/core/generic/grpc_plugin_dll_class.py#L292) and it seems to be interfering with how `pytomlpp` works as we are no longer able to load a TOML file. See
image
2. `toml` does provide **read and write** capabilities while `tomli` (a more standard package) does only provide read capabilities.

---

> **NOTE**: AFAIK `mono` needs to be reachable (hence the current LIBRARY_PATH extension) for `pyedb` to be able to load `EDBWrapper`. Otherwise, instantiating an `Edb` object will raise an exception. This can be avoided if one installs `mono` manually but I'm not chaning this as it is probably a standard work around used by our users. At some point, it would be wise to remove `dotnetcore/dotnetcore2` and ensure that `dotnet` + `mono` are installed if required.

### CODE REVIEW

**Comment by SMoraisAnsys** [comment]
@svandenb-dev This PR won't be able to pass until a new release of `pyedb`.

**Review by Samuelopez-ansys — APPROVED** [review]

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -81,14 +81,15 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ env.MAIN_PYTHON_VERSION }}
+          use-python-cache: false
```
Comment (SMoraisAnsys) [inline .github/workflows/ci_cd.yml:84]:
```suggestion
```

File: .github/workflows/ci_cd.yml [hunk 2]
```diff
@@ -81,14 +81,15 @@ jobs:
...
 
       - name: Update pip
         run: |
           pip install --upgrade pip
 
       - name: Install pyaedt and documentation dependencies
         run: |
-          pip install .[doc-no-examples]
+          pip install .[doc-no-examples] --no-cache
```
Comment (SMoraisAnsys) [inline .github/workflows/ci_cd.yml:92]:
```suggestion
          pip install .[doc-no-examples]
```

File: .github/workflows/ci_cd.yml [hunk 3]
```diff
@@ -160,7 +161,7 @@ jobs:
       - name: Install pyaedt and documentation dependencies
         run: |
           .venv\Scripts\Activate.ps1
-          pip install .[doc]
+          pip install .[doc] --no-cache
```
Comment (SMoraisAnsys) [inline .github/workflows/ci_cd.yml:164]:
```suggestion
          pip install .[doc]
```

File: .github/workflows/ci_cd.yml [hunk 4]
```diff
@@ -237,6 +238,7 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ env.MAIN_PYTHON_VERSION }}
+          use-python-cache: false
```
Comment (SMoraisAnsys) [inline .github/workflows/ci_cd.yml:241]:
```suggestion
```

File: .github/workflows/ci_cd.yml [hunk 5]
```diff
@@ -249,7 +251,7 @@ jobs:
       - name: Install pyaedt and tests dependencies
         run: |
           .venv\Scripts\Activate.ps1
-          pip install .[tests]
+          pip install .[tests] --no-cache
```
Comment (SMoraisAnsys) [inline .github/workflows/ci_cd.yml:254]:
```suggestion
          pip install .[tests]
```

File: .github/workflows/ci_cd.yml [hunk 6]
```diff
@@ -300,6 +302,7 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ env.MAIN_PYTHON_VERSION }}
+          use-python-cache: false
```
Comment (SMoraisAnsys) [inline .github/workflows/ci_cd.yml:305]:
```suggestion
```

File: .github/workflows/ci_cd.yml [hunk 7]
```diff
@@ -314,7 +317,7 @@ jobs:
         run: |
           export LD_LIBRARY_PATH=${{ env.ANSYSEM_ROOT242 }}/common/mono/Linux64/lib64:${{ env.ANSYSEM_ROOT242 }}/Delcross:$LD_LIBRARY_PATH
           source .venv/bin/activate
-          pip install .[tests]
+          pip install .[tests] --no-cache
```
Comment (SMoraisAnsys) [inline .github/workflows/ci_cd.yml:320]:
```suggestion
          pip install .[tests]
```

File: .github/workflows/ci_cd.yml [hunk 8]
```diff
@@ -353,6 +356,7 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ env.MAIN_PYTHON_VERSION }}
+          use-python-cache: false
```
Comment (SMoraisAnsys) [inline .github/workflows/ci_cd.yml:359]:
```suggestion
```

File: .github/workflows/ci_cd.yml [hunk 9]
```diff
@@ -365,7 +369,7 @@ jobs:
       - name: Install pyaedt and tests dependencies
         run: |
           .venv\Scripts\Activate.ps1
-          pip install .[tests]
+          pip install .[tests] --no-cache
```
Comment (SMoraisAnsys) [inline .github/workflows/ci_cd.yml:372]:
```suggestion
          pip install .[tests]
```

File: .github/workflows/ci_cd.yml [hunk 10]
```diff
@@ -421,6 +425,7 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ env.MAIN_PYTHON_VERSION }}
+          use-python-cache: false
```
Comment (SMoraisAnsys) [inline .github/workflows/ci_cd.yml:428]:
```suggestion
```

File: .github/workflows/ci_cd.yml [hunk 11]
```diff
@@ -435,7 +440,7 @@ jobs:
         run: |
           export LD_LIBRARY_PATH=${{ env.ANSYSEM_ROOT242 }}/common/mono/Linux64/lib64:${{ env.ANSYSEM_ROOT242 }}/Delcross:$LD_LIBRARY_PATH
           source .venv/bin/activate
-          pip install .[tests]
+          pip install .[tests] --no-cache
```
Comment (SMoraisAnsys) [inline .github/workflows/ci_cd.yml:443]:
```suggestion
          pip install .[tests]
```

File: _unittest/test_04_SBR.py [hunk 12]
```diff
@@ -102,7 +102,8 @@ def init(self, aedtapp, local_scratch):
         self.local_scratch = local_scratch
         if not is_linux:
             # this should be changed upstream to use a HOME or TEMP folder by default...
-            osmnx.settings.cache_folder = os.path.join(local_scratch.path, "cache")
+            # osmnx.settings.cache_folder = os.path.join(local_scratch.path, "cache")
+            pass
```
Comment (SMoraisAnsys) [inline _unittest/test_04_SBR.py:106]:
```suggestion
            osmnx.settings.cache_folder = os.path.join(local_scratch.path, "cache")
```

File: _unittest/test_04_SBR.py [hunk 13]
```diff
@@ -241,15 +242,15 @@ def test_11_add_sbr_boundaries_in_hfss_solution(self, add_app):
...
-            assert os.path.exists(parts_dict["parts"][part]["file_name"])
+    # @pytest.mark.skipif(is_linux, reason="Not supported.")
+    # def test_12_import_map(self):
+    #     self.aedtapp.insert_design("city")
+    #     ansys_home = [40.273726, -80.168269]
+    #     parts_dict = self.aedtapp.modeler.import_from_openstreet_map(
+    #         ansys_home, terrain_radius=200, road_step=3, plot_before_importing=False, import_in_aedt=True
+    #     )
+    #     for part in parts_dict["parts"]:
+    #         assert os.path.exists(parts_dict["parts"][part]["file_name"])
```
Comment (SMoraisAnsys) [inline _unittest/test_04_SBR.py:253]:
```suggestion
    @pytest.mark.skipif(is_linux, reason="Not supported.")
    def test_12_import_map(self):
        self.aedtapp.insert_design("city")
        ansys_home = [40.273726, -80.168269]
        parts_dict = self.aedtapp.modeler.import_from_openstreet_map(
            ansys_home, terrain_radius=200, road_step=3, plot_before_importing=False, import_in_aedt=True
        )
        for part in parts_dict["parts"]:
            assert os.path.exists(parts_dict["parts"][part]["file_name"])
```

File: pyproject.toml [hunk 14]
```diff
@@ -49,9 +49,9 @@ tests = [
     "joblib>=1.0.0,<1.5",
     "matplotlib>=3.5.0,<3.10",
     "mock>=5.1.0,<5.2",
-    "numpy>=1.20.0,<2",
+    "numpy>=1.20.0,<2.2",
     "openpyxl>=3.1.0,<3.3",
-    "osmnx>=1.1.0,<1.10",
+    # "osmnx>=1.1.0,<1.10",
```
Comment (SMoraisAnsys) [inline pyproject.toml:54]:
```suggestion
    "osmnx>=1.1.0,<1.10",
```

File: pyproject.toml [hunk 15]
```diff
@@ -79,7 +79,7 @@ doc = [
     "nbsphinx>=0.9.0,<0.10",
     "numpydoc>=1.5.0,<1.9",
     "openpyxl>=3.0.0,<3.2",
-    "osmnx>=1.1.0,<1.10",
+    # "osmnx>=1.1.0,<1.10",
```
Comment (SMoraisAnsys) [inline pyproject.toml:82]:
```suggestion
    "osmnx>=1.1.0,<1.10",
```

File: pyproject.toml [hunk 16]
```diff
@@ -116,9 +116,9 @@ doc-no-examples = [
 ]
 all = [
     "matplotlib>=3.5.0,<3.10",
-    "numpy>=1.20.0,<2",
+    "numpy>=1.20.0,<2.2",
     "openpyxl>=3.1.0,<3.3",
-    "osmnx>=1.1.0,<1.10",
+    # "osmnx>=1.1.0,<1.10",
```
Comment (SMoraisAnsys) [inline pyproject.toml:121]:
```suggestion
    "osmnx>=1.1.0,<1.10",
```

File: pyproject.toml [hunk 17]
```diff
@@ -130,9 +130,9 @@ all = [
 ]
 installer = [
     "matplotlib>=3.5.0,<3.10",
-    "numpy>=1.20.0,<2",
+    "numpy>=1.20.0,<2.2",
     "openpyxl>=3.1.0,<3.3",
-    "osmnx>=1.1.0,<1.10",
+    # "osmnx>=1.1.0,<1.10",
```
Comment (SMoraisAnsys) [inline pyproject.toml:135]:
```suggestion
    "osmnx>=1.1.0,<1.10",
```

**Review by SMoraisAnsys — COMMENTED** [review]
To be added after success of the CI workflow.

**Review by Samuelopez-ansys — APPROVED** [review]

**Comment by SMoraisAnsys** [comment]
Seems like the fix is not working when adding back `osmnx` and merging main. I need to double check if this is related to using `numpy<2` due to `osmnx`.

**Pull request closed without merging**